core/ledger/kvledger/snapshot_mgmt.go

```diff
@@ -60,3 +62,130 @@ func (l *kvLedger) ListSnapshots() ([]string, error) {
 func (l *kvLedger) DeleteSnapshot(height uint64) error {
 	return errors.Errorf("not implemented")
 }
+
+// snapshotRequestBookkeeper manages snapshot requests in a leveldb and maintains smallest height for pending snapshot requests
+type snapshotRequestBookkeeper struct {
+	dbHandle              *leveldbhelper.DBHandle
+	smallestRequestHeight uint64
+}
+
+func newSnapshotRequestBookkeeper(dbHandle *leveldbhelper.DBHandle) (*snapshotRequestBookkeeper, error) {
+	bk := &snapshotRequestBookkeeper{dbHandle: dbHandle}
+
+	var err error
+	if bk.smallestRequestHeight, err = bk.smallestRequest(); err != nil {
+		return nil, err
+	}
+
+	return bk, nil
+}
+
+// add adds the given height to the bookkeeper db and returns an error if the height already exists
+func (bk *snapshotRequestBookkeeper) add(height uint64) error {
```

Comment:
Nit picking: Idiomatically, you would use one letter name for the function receiver. You may like to change `bk` to `b` or `k`